### PR TITLE
[Fix bug 1247219] UI fixes for new group settings

### PR DESCRIPTION
**Pull request by comzeradd**
* Minimize space on create group modal
* Use even space on no pending invitations text
* Minimize space on group types descriptions
* Add markdown informative text on terms
* Fix group access criteria toggle

### CODE REVIEW

**Comment by comzeradd** [comment]
@johngian r?

File: mozillians/jinja2/groups/includes/group_edit_invitation.html [hunk 1]
```diff
@@ -41,7 +41,7 @@
               </div>
             {% endfor %}
           {% else %}
-            <p> This group has no pending invitations.</p>
+            <p class="text-muted">This group has no pending invitations.</p>
```
Comment (johngian) [inline mozillians/jinja2/groups/includes/group_edit_invitation.html]: ``text-muted`` looks OK. I suggest you also add a CSS rule like the following (with a more appropriate selector):
```
form.edit-group p {
    text-align: center;
}
```

**Comment by johngian** [comment]
New membership terms and membership expiration have different label text format (hint ``:``)

**Comment by johngian** [comment]
I still suggest we consolidate terms/expiration. It doesn't really make any sense to have 2 different panels for these.

@akatsoulas what do you think?

File: mozillians/jinja2/groups/includes/group_edit_invitation.html (hunk 1 above)
Comment (comzeradd) [inline mozillians/jinja2/groups/includes/group_edit_invitation.html]: I'm not sure about the alignment. We have no text aligned in center. I think it will look a bit odd.

**Comment by comzeradd** [comment]
I agree. This would require mostly a backend forms consolidation, rather that frontend changes.

File: mozillians/jinja2/groups/includes/group_edit_invitation.html (hunk 1 above)
Comment (johngian) [inline mozillians/jinja2/groups/includes/group_edit_invitation.html]: It looks a bit odd to me now. It feels like its a common layout to have centered informative text when a list is empty.

**Comment by johngian** [comment]
I think that the way we handle forms right now in the backend it would be the exact same thing if we post these 2 fields seperately and together.

File: mozillians/jinja2/groups/includes/group_edit_access.html [hunk 2]
```diff
@@ -25,7 +25,7 @@
           </ul>
         {% endif %}
 
-        <fieldset id="access">
+        <fieldset id="access" class="last-fieldset">
```
Comment (johngian) [inline mozillians/jinja2/groups/includes/group_edit_access.html]: Why do we need this class? Also why all that DOM magic when you could select the last fieldset using CSS selectors?

File: mozillians/static/mozillians/js/group_edit.js [hunk 3]
```diff
@@ -18,8 +19,10 @@ $(function() {
     function checkCriteria(modVal) {
         if(modVal === 'by_request') {
             $id_new_member_criteria_fieldset.show();
+            $access.removeClass('last-fieldset');
```
Comment (johngian) [inline mozillians/static/mozillians/js/group_edit.js]: Same here.

File: mozillians/static/mozillians/js/group_edit.js [hunk 4]
```diff
@@ -18,8 +19,10 @@ $(function() {
...
         } else {
             $id_new_member_criteria_fieldset.hide();
+            $access.addClass('last-fieldset');
```
Comment (johngian) [inline mozillians/static/mozillians/js/group_edit.js]: Same here.

File: mozillians/static/mozillians/js/group_edit.js [hunk 5]
```diff
@@ -42,10 +45,11 @@ $(function() {
     }
 
     // Hide/show field when moderation field changes
-    $id_accepting_new_members.change(function() {
-        checkCriteria($id_accepting_new_members.val());
+    $('#form-access input[type="radio"]').on('change', function() {
```
Comment (johngian) [inline mozillians/static/mozillians/js/group_edit.js]: Please use a var for that. Ideally we should use an ``id``.

**Comment by johngian** [comment]
Membership expiration js is also broken.

**Comment by johngian** [comment]
New membership criteria textfield should be under the label like in the rest of the same fields

selection_010

File: mozillians/static/mozillians/js/group_edit.js [hunk 6]
```diff
@@ -42,10 +45,11 @@ $(function() {
...
+        $accepting_new_members = $('input[name=accepting_new_members]:checked');
+        checkCriteria($accepting_new_members.val());
     });
-    $id_membership_can_expire.change(function() {
+    $accepting_new_members.change(function() {
```
Comment (johngian) [inline mozillians/static/mozillians/js/group_edit.js]: This line here is wrong. You are checking group type to figure out if group membership expires.

Comment (akatsoulas) [inline mozillians/static/mozillians/js/group_edit.js]: Why is this changed in the first place?

**Comment by akatsoulas** [comment]
:+1: for merging the two sections into one.

File: mozillians/jinja2/groups/includes/group_edit_access.html (hunk 2 above)
Comment (comzeradd) [inline mozillians/jinja2/groups/includes/group_edit_access.html]: The `:last-child` selector doesn't work here because this is not considered last child, even when the next fieldset is hidden.

File: mozillians/static/mozillians/js/group_edit.js (hunk 6 above)
Comment (comzeradd) [inline mozillians/static/mozillians/js/group_edit.js]: Because the previous code was broken. Terms input text is never revealed.

`$accepting_new_members` is `'input[name=accepting_new_members]:checked'`, which just checks which of all three radio buttons is checked.

Comment (johngian) [inline mozillians/static/mozillians/js/group_edit.js]: Again this is for the group type not for the membership expiration.

File: mozillians/jinja2/groups/includes/group_edit_invitation.html (hunk 1 above)
Comment (comzeradd) [inline mozillians/jinja2/groups/includes/group_edit_invitation.html]: We don't use that kind of alignment nowhere else in the site, afaik. Only on alerts.

Comment (johngian) [inline mozillians/jinja2/groups/includes/group_edit_invitation.html]: Do we have a similar layout somewhere in Mozillians.org so we can follow the same paradigm?

Comment (johngian) [inline mozillians/jinja2/groups/includes/group_edit_invitation.html]: If not I suggest we go with the centered text.

Comment (comzeradd) [inline mozillians/jinja2/groups/includes/group_edit_invitation.html]: We have cases where data is pending (like secondary emails or api keys), but not a 100% similar case with this. We just have no centered text in any settings option or text.

Comment (johngian) [inline mozillians/jinja2/groups/includes/group_edit_invitation.html]: I suggest we go with the centered and let's reevaluate it in the future.

**Comment by comzeradd** [comment]
PR updated. Please test that everything works as expected, especially at the Access tab.

**Comment by johngian** [comment]
Please reduce the space between the 2 checkboxes:
selection_013

**Comment by johngian** [comment]
Also do we need these 2 labels to be bold? (no strong opinion on that)

**Comment by johngian** [comment]
I preferred the previous UI with the choices after the label.
selection_014

**Comment by johngian** [comment]
Also you need to rebase and adapt the code after #1391 fixes.

**Comment by johngian** [comment]
"Membership can expire" is not working.

* Clicked checkbox
* Added days

No data gets saved.

**Comment by comzeradd** [comment]
I fixed the modal misplacement and rebased with master.

I reverted the change with joining terms and expiration. This doesn't make sense to be part of this PR. The forms are separated in the backend and even the tests have to be rewritten.

File: mozillians/static/mozillians/js/group_edit.js [hunk 7]
```diff
@@ -7,9 +7,10 @@ $(function() {
         }
     }
 
-    $id_accepting_new_members = $('#id_accepting_new_members');
+    $group_type_choice = $('#form-access input[type="radio"]');
+    $accepting_new_members = $('input[name=accepting_new_members]:checked');
     $id_new_member_criteria_fieldset = $('#id_new_member_criteria_fieldset');
-    $id_invalidation_days = $('#id_group_invalidation_days');
+    $id_group_invalidation_days = $('#id_group_invalidation_days');
```
Comment (johngian) [inline mozillians/static/mozillians/js/group_edit.js]: No need to rename the vars here.

File: mozillians/static/mozillians/js/group_edit.js [hunk 8]
```diff
@@ -42,8 +43,9 @@ $(function() {
     }
 
     // Hide/show field when moderation field changes
-    $id_accepting_new_members.change(function() {
-        checkCriteria($id_accepting_new_members.val());
+    $($group_type_choice).on('change', function() {
```
Comment (johngian) [inline mozillians/static/mozillians/js/group_edit.js]: No need to reselect the selection.

**Comment by comzeradd** [comment]
@johngian PR updated

File: mozillians/static/mozillians/js/group_edit.js [hunk 9]
```diff
@@ -42,8 +43,9 @@ $(function() {
     }
 
     // Hide/show field when moderation field changes
-    $id_accepting_new_members.change(function() {
-        checkCriteria($id_accepting_new_members.val());
+    $group_type_choice.on('change', function() {
+        $accepting_new_members = $('input[name=accepting_new_members]:checked');
```
Comment (johngian) [inline mozillians/static/mozillians/js/group_edit.js]: Instead of reselecting this here you could use ``this``.

Comment (johngian) [inline mozillians/static/mozillians/js/group_edit.js]: Instead of copy/pasting the selector here you could reevaluate like:
``$accepting_new_members = $($accepting_new_members.selector)``

**Comment by johngian** [comment]
Code looks OK. r+wc

**Comment by johngian** [comment]
Please test it locally before you merge.
